modules/core/src/main/java/org/apache/ignite/internal/processors/metastorage/persistence/DistributedMetaStorageImpl.java

```diff
@@ -1210,6 +1210,39 @@ private void onAckMessage(
         }
     }
 
+    /**
+     * @return Future which will be completed when all the updates prior to the pause processed.
+     */
+    public Future<?> flush() {
+        assert isPersistenceEnabled;
+
+        lock.readLock().lock();
+
+        try {
+            return worker.flush();
+        }
+        finally {
+            lock.readLock().unlock();
+        }
+    }
+
+    /**
+     * @param compFut Future which should be completed when worker may proceed with updates.
+     */
+    public void pauseMetaStorage(IgniteInternalFuture<?> compFut) {
```

Comment:
maybe it would be better to name this method ``suspend`` (or at least remove duplicate ``MetaStorage`` (metastorage.pauseMetaStorage(...))

Reply:
Fixed.